src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidTestCommand.cs

```diff
@@ -38,6 +39,20 @@ protected override Task<ExitCode> InvokeInternal(ILogger logger)
             }
             var runner = new AdbRunner(logger);
 
+            // Assumption: APKs we test will only have one arch for now
+            string apkRequiredArchitecture = string.IsNullOrEmpty(_arguments.DeviceArchitecture) ?
+                ApkHelper.GetApkSupportedArchitectures(_arguments.AppPackagePath).FirstOrDefault() :
+                _arguments.DeviceArchitecture;
+
+            if (!string.IsNullOrEmpty(_arguments.DeviceArchitecture))
+            {
+                logger.LogInformation($"Will attempt to run device on specified architecture: '{apkRequiredArchitecture}'");
```

Comment:
nit: if we have this `if/else` already, would maybe make sense to fill `apkRequiredArchitecture` in here too. Now we have two different conditional blocks for the same thing